### PR TITLE
Move existing duplicated code from GammaCorrection() into on function

### PR DESCRIPTION
**Pull request by gumeo**
This should solve issue #545.

### CODE REVIEW

File: src/cuda/image_function_cuda.cu [hunk 1]
```diff
@@ -770,18 +770,7 @@ namespace Image_Function_Cuda
...
-        // We precalculate all values and store them in lookup table
-        std::vector < uint8_t > value( 256, 255u );
-
-        for ( uint16_t i = 0; i < 256; ++i ) {
-            double data = a * pow( i / 255.0, gamma ) * 255 + 0.5;
-
-            if ( data < 256 )
-                value[i] = static_cast<uint8_t>(data);
-        }
+	std::vector<uint8_t> value = GetGammaCorrectionLookupTable( a, gamma );
```
Comment (ihhub) [inline src/cuda/image_function_cuda.cu]: Please use 4 space indentation instead of tabs.

Comment (ihhub) [reply src/cuda/image_function_cuda.cu]: `const std::vector<uint8_t> & value = Image_Function_Helper::GetGammaCorrectionLookupTable( a, gamma );`

File: src/image_function_helper.cpp [hunk 2]
```diff
@@ -389,6 +389,23 @@ namespace Image_Function_Helper
         return out;
     }
 
+    std::vector<uint8_t> GetGammaCorrectionLookupTable( double a, double gamma ){
```
Comment (ihhub) [inline src/image_function_helper.cpp]: Please put `{` on separate line.

File: src/image_function_helper.cpp [hunk 3]
```diff
@@ -389,6 +389,23 @@ namespace Image_Function_Helper
...
+        std::vector < uint8_t > value( 256, 255u );
+
+        for ( uint16_t i = 0; i < 256; ++i ) {
+            double data = a * pow( i / 255.0, gamma ) * 255 + 0.5;
+
+            if ( data < 256 )
+                value[i] = static_cast<uint8_t>(data);
+        }
+
+	return value;
```
Comment (ihhub) [inline src/image_function_helper.cpp]: Please use space indentation.

File: src/image_function.cpp [hunk 4]
```diff
@@ -811,18 +811,7 @@ namespace Image_Function
...
-        // We precalculate all values and store them in lookup table
-        std::vector < uint8_t > value( 256, 255u );
-
-        for( uint16_t i = 0; i < 256; ++i ) {
-            double data = a * pow( i / 255.0, gamma ) * 255 + 0.5;
-
-            if( data < 256 )
-                value[i] = static_cast<uint8_t>(data);
-        }
+        std::vector<uint8_t> value = GetGammaCorrectionLookupTable( a, gamma );
```
Comment (ihhub) [inline src/image_function.cpp]: Please use `const std::vector<uint8_t> & value = Image_Function_Helper::GetGammaCorrectionLookupTable( a, gamma );`

**Comment by gumeo** [comment]
@ihhub thanks for the comments, can you please review if I missed something?

**Comment by gumeo** [comment]
I added the `cmath` include where `pow` was used. Not sure if you prefer it in the header file or not.

**Comment by gumeo** [comment]
@ihhub I don't understand the logs from the travis-ci job, do you spot what is missing?

**Comment by ihhub** [comment]
@gumeo thank you very much for the contribution to the project!

**Pull request merged**